tests/jobs/test_scheduler_job.py

```diff
@@ -645,6 +645,89 @@ def test_find_executable_task_instances_in_default_pool(self, dag_maker):
         session.rollback()
         session.close()
 
+    def test_max_active_task_in_a_dag_with_large_task_does_not_affect_other_dags(self, dag_maker, session):
+        dag_id_1 = 'first_dag'
+        dag_id_2 = 'second_dag'
+        with dag_maker(dag_id_1, max_active_tasks=1):
+            for i in range(10):
+                BashOperator(task_id=f"mytask{i}", bash_command='sleep 1d')
+        dr1 = dag_maker.create_dagrun(run_id=dag_id_1, state=DagRunState.RUNNING)
+        tis1 = dr1.get_task_instances()
+        for ti in tis1:
+            ti.state = TaskInstanceState.SCHEDULED
+            session.merge(ti)
+        session.flush()
+        self.scheduler_job = SchedulerJob(subdir=os.devnull)
+        res = self.scheduler_job._executable_task_instances_to_queued(max_tis=32, session=session)
+        assert 1 == len(res)
+        assert session.query(TaskInstance).filter(TaskInstance.state == TaskInstanceState.QUEUED).count() == 1
+        self.scheduler_job._executable_task_instances_to_queued(max_tis=32, session=session)
+        session.flush()
+        # assert last_scheduling_decision is updated
+        assert (
+            session.query(TaskInstance)
+            .filter(TaskInstance.state == State.SCHEDULED, TaskInstance.last_scheduling_decision.__ne__(None))
```

Comment:
Had to use `__ne__` because `is_not` was not working(I think cos of our sqlalchemy version)

Reply:
https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.ColumnOperators.__ne__

>In a column context, produces the clause a != b. If the target is None, produces a IS NOT NULL.

I am surprised it didn't work

Reply:
Maybe it's not available in SQLAlchemy==1.3.24 which we use

Reply:
I think `isnot()` works in all 1.x versions.